flowershop/models/stock_lot.py

```diff
@@ -0,0 +1,54 @@
+from odoo import fields,models ,api
+from odoo.exceptions import ValidationError
+
+
+
+class stocklot(models.Model):
+    _inherit = "stock.lot"
+
+    water_ids = fields.One2many("flower_water", "serial_id")
+
+    is_flower=fields.Boolean(related="product_id.is_flower")
+
+
+    def action_water_flower(self):
+        flowers = self.filtered(lambda rec: rec.is_flower)
+
+        for record in flowers:
+            if record.water_ids:
+                last_watered_date = record.water_ids[0].date
+                frequency = record.product_id.flower_ids.watering_frq
+                today = fields.Date.today()
+                if (today - last_watered_date).days < frequency:
+                    raise ValidationError(f"Flower {record.name} Watered Recently!")
+                    continue
+            self.env["flower_water"].create({
+            "serial_id": record.id,
+            "date":fields.Date.today() })
```

Comment:
That  is fine, it works. However the issue here is whenever we raise errors in odoo it stops the execution of the program. meaning nothing is saved to the db.

A better approach would be to store every bad value in a list and after committing raise an error. 

Check the  solution in the main branch to see how it is done. 

This is not a mistake, just a better approach.